plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -291,6 +291,17 @@ public static Object convert(Object value, Type type)
             // TODO the client sees the bytearray's tostring ouput instead of seeing actual bytes, needs to be fixed.
             return ((ByteBuffer) value).array();
         }
+        if (type instanceof Types.IntegerType || type instanceof Types.DateType) {
```

Comment:
You can break both cases into their own branches since the `instanceof Long` should not be happening for Integer type. The current code feels a bit misleading.

Same for DateType - the underlying value is either going to be Integer or Long - it cannot be both.

Any cases where the alternative branch gets taken?

Reply:
+1